docs/pages/dev/how_to/planning-pipeline-integration/planning-pipeline-integration.en.md

```diff
@@ -276,29 +319,116 @@ def test_connection(connection_host: str, connection_token: str) -> bool:
 
 The pipeline integration uses a dual approach for task status management:
 
-#### 1. Pipeline Progress Updates (Optional)
+#### 1. Pipeline Progress Updates (Recommended)
 
-The pipeline can optionally update progress and messages using the generic API caller:
+The pipeline uses a `TaskProgressLogger` class for convenient progress tracking:
 
 ```python
-def safe_update_task_status(
-    task_id: Optional[int],
-    message: str,
+class TaskProgressLogger:
+    """Logger for task progress with Iaso integration.
+    
+    Encapsulates task context to avoid repetitive parameter passing.
+    Implements smart batching to reduce API calls.
+    """
+
+    def __init__(
+        self,
+        task_id: Optional[int],
+        pipeline_id: str,
+        connection_host: str,
+        connection_token: str,
+        batch_threshold: int = 5,
+    ):
+        """Initialize the logger with task context.
+        
+        Args:
+            task_id: ID of the task (can be None)
+            pipeline_id: Pipeline ID for the endpoint
+            connection_host: Iaso server URL
+            connection_token: Bearer token for authentication
+            batch_threshold: Minimum progress change to trigger API call (default: 5%)
+        """
+
+    def log(
+        self,
+        message: str,
+        progress_value: Optional[int] = None,
+        result_data: Optional[Dict] = None,
+        log_level: str = "info",
+        force_send: bool = False,
+    ):
+        """Log progress with current task context.
+        
+        Args:
+            message: Progress message
+            progress_value: Optional progress value (end_value is always 100)
+            result_data: Optional result data to save (usually for final step)
+            log_level: Level of the log (info, error, warning, debug)
+            force_send: Force immediate API call (bypasses batching)
+        """
+
+    def flush(self):
+        """Force send any pending updates (call at end of pipeline)."""
+```
```

Comment:
You shouldn't have to copy the source code here:
1. It makes the doc harder to maintain
2. Our doc engine is able to generate a doc page from the class and the python docstring, so only the code and the comments in the code need updating